src/internal/connector/graph/service_helper.go

```diff
@@ -94,31 +94,51 @@ func (handler *LoggingMiddleware) Intercept(
 	}
 
 	if (resp.StatusCode / 100) == 2 {
-		return resp, err
-	}
-
-	// special case for supportability: log all throttling cases.
-	if resp.StatusCode == http.StatusTooManyRequests {
-		logger.Ctx(ctx).Infow("graph api throttling", "method", req.Method, "url", req.URL)
-	}
+		if logger.DebugAPI || os.Getenv(logGraphRequestsEnvKey) != "" {
```

Comment:
Let's dump response here (like line 114) but without the body i.e. httputil.DumpResponse(resp, false)